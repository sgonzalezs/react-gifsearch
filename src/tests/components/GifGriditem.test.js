import React from 'react';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';



describe('Tests file GifGridItem.js', () => {
    
    const title='Category Title';
    const url='http://localhost/category-image.jpg';
    const wrapper=shallow(<GifGridItem url={url} title={title} />);

    test('load Component', () => {
        
        expect(wrapper).toMatchSnapshot();
    });

    test('must contain a text', () => {

        const p=wrapper.find('p');
        expect(p.text().trim()).toBe(title);
        
    });

    test('must contain an image with url and title alt', () => {

        const img=wrapper.find('img')

        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    });

    test('must contain the class animate__fadeIn', () => {
        
        const div=wrapper.find('div');
        
        const class_animate=div.prop('className').split(" ")[2];
        const className=div.prop('className');

        expect(class_animate).toBe('animate__fadeIn');
        expect(className.includes('animate__fadeIn')).toBe(true);
        
    });
    
});
