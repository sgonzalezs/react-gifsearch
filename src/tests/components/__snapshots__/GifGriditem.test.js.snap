// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests file GifGridItem.js load Component 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Category Title"
    src="http://localhost/category-image.jpg"
  />
  <p>
    Category Title
  </p>
</div>
`;
